### PR TITLE
Pinter added as a demo.

### PR DESCRIPTION
**Pull request by VoxWave**

### CODE REVIEW

**Comment by WaDelma** [comment]
Pinter :D

**Comment by VoxWave** [comment]
Yes it's pinter. Because I typoed printer ages ago when I made this.

File: demos/pinter/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    let mut p = PathBuf::new();
+    p.push(name);
+    p.set_extension("png");
+    level_image.save(p.as_path()).expect("Something went wrong when saving the png.");
+}
+
+fn faction_to_color(colors: &HashMap<Faction, Rgb<u8>>, tile: &Result<&Faction, Error>) -> Rgb<u8> {
+    match *tile {
+        Ok(ref tile) => {
+                match colors.get(tile) {
```
Comment (WaDelma) [inline demos/pinter/src/main.rs]: Indentation?

File: demos/pinter/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    Some(color) => color.clone(),
+                    None => Rgb([0u8, 0u8, 0u8]),
```
Comment (WaDelma) [inline demos/pinter/src/main.rs]: There is faction of this color.

File: demos/pinter/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                }
+        },
+        Err(Error::IndexOutOfBounds) => panic!("Tile png import failed. Level indexing went out of bounds."),
+    }
+}
+
+fn usize_from_cmd() -> usize {
```
Comment (WaDelma) [inline demos/pinter/src/main.rs:96]: Refactor `usize` and `u64` parsing? Maybe just have usize parsing?

File: demos/pinter/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    for i in 0..iterations {
+        factiongen.generate(&mut level, &mut buffer);
+        println!("{} iterations done.", i);
+    }
+    faction_png_export(String::from("picture"), &colors, &mut level);
+}
+
+fn populate_level(level: &mut Level<Faction>, rangemax: usize) {
+    let range = Range::new(0,rangemax);
+    let mut rand = XorShiftRng::rand(&mut OsRng::new().unwrap());
```
Comment (WaDelma) [inline demos/pinter/src/main.rs:58]: This means you cannot replay the same generation... Maybe take seed from user.

**Review by WaDelma — COMMENTED** [review]
PINTER

File: demos/pinter/src/main.rs (hunk 2 above)
Comment (VoxWave) [reply demos/pinter/src/main.rs]: Does it matter? The whole thing should be populated with... oh right. I can just make it panic in the none case can't I.

File: demos/pinter/src/main.rs (hunk 3 above)
Comment (VoxWave) [reply demos/pinter/src/main.rs:96]: umm... I wonder why I used a u64 for the iterations.

File: demos/pinter/src/main.rs (hunk 2 above)
Comment (WaDelma) [reply demos/pinter/src/main.rs]: There is none case right? When you start the whole level is not filled?
Or am I mistaken?

Comment (WaDelma) [reply demos/pinter/src/main.rs]: Or am I mistaken?

File: demos/pinter/src/main.rs (hunk 4 above)
Comment (VoxWave) [reply demos/pinter/src/main.rs:58]: this we can do later.

File: demos/pinter/src/main.rs (hunk 2 above)
Comment (VoxWave) [reply demos/pinter/src/main.rs]: the whole level is filled

**Review by WaDelma — APPROVED** [review]

**Pull request merged**